src/EditorFeatures/Test/CommentSelection/CSharpToggleBlockCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,315 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Editor.CSharp.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.Implementation.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Operations;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.CommentSelection
+{
+    [UseExportProvider]
+    public class CSharpToggleBlockCommentCommandHandlerTests : AbstractToggleBlockCommentTestBase
+    {
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleBlockComment)]
+        public void AddComment_CaretInsideToken()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        va$$r i = 1;
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+        var/**/ i = 1;
+    }
+}";
+
+            var expectedSelectedSpans = new[]
+            {
+                Span.FromBounds(46, 50)
```

Comment:
ick.  Ideall avoid thsi.  We have ways to write tests taht can include spans in them.  So our tests can just be markup and your testing method will validate it.

Reply:
the benefit here is that these numbers are meaningless.  if you put this in the markup, then it's clear which spans you were expecting to be selected, and a reviewer can tell if this is sensible or not.  right now i have no idea if this is appropriate or not.